functions.py

```diff
@@ -1,34 +1,86 @@
 import time
+SEC_IN_YEAR = 31536000.0
 
-def ReadLines(nameoffile):
+#function returns data fron text in list(array)
+def readStateFile(nameoffile):
     with open(nameoffile,'r') as f:
-        arrayline = f.readlines()
-        return arrayline
+        arrdata = f.readlines()
+        if arrdata != None:
+            return arrdata
+        else:
+            print("Error, empty file, try another one")
+            return 0
+        
+#check the order by date        
+def checkOrderByDate(arrdata):
+    previousdate = 0.0
+    for i in range(0, len(arrdata)):
+        arrline  = arrdata[i].split(',', 2)
+        if float(arrline[1]) < previousdate:
+            print("array doesnt ordered by date")
+            print("previousdate:", previousdate, "date:", arrline[1]) 
+            return 1
+        previousdate = float(arrline[1])
+    return 0
+
+# sorting by time btw
+def SortByDate(strings):
+    sorted_strings = sorted(strings, key=lambda line: float(line.split(",")[1]))
+    return sorted_strings
 
 #how much is not taxed
-def SellWithoutTax():
-    arrayline = ReadLines('new_register.txt')
+def SellWithoutTax(arrdata):
     sum = 0.0
-    i = 0
-    arr = arrayline[i].split(',', 2)
-    while (time.time() - float(arr[1])) >= 31536000.0:
-        sum += float(arr[0])
-        i += 1
-        arr = arrayline[i].split(',', 2)
+    for i in range(0, len(arrdata)):
+        arrline  = arrdata[i].split(',', 2)
+        if (time.time() - float(arrline[1])) >= SEC_IN_YEAR:

```

Comment:
Remember there are 2 conditions for BTC to be sold without tax.
One is held more than a year and the other is that no profit is made.
I.e. IF price is now 16K But I bought it at 30K I can definitely sell int without taxes.

But be careful if say now price is 16K, but I bought 1BTC for 30K 6 month ago and then 1BTC for 8K 2 months ago I still don't want to count the ones I bought or 30K. Cause I cannot sell them before I sell the ones for 8K.